beartype_test/a90_func/package/test_package_import.py

```diff
@@ -48,9 +52,14 @@ def test_package_import_isolation() -> None:
         r'beartype\.cave',
         r'beartype\.vale',
         r'numpy',
-        r'typing_extensions',
     )
 
+    # Add typing_extensions to the list if we're outside 3.7
+    inside_3_7 = IS_PYTHON_AT_LEAST_3_7 and not IS_PYTHON_AT_LEAST_3_8
+    outside_3_7 = not inside_3_7
+    if outside_3_7:
+        _HEAVY_MODULE_NAME_RAW_REGEXES += (r'typing_extensions',)
```

Comment:
This is a compromise to address https://github.com/beartype/beartype/pull/103/files#r814527877.